File: week3/lab_py/car.py
class Car:
    def __init__(self) -> None:
        self.make = "BMW"
        self.pos = 0

    def move(self,distance):
        self.pos += distance

    def __str__(self):
        return f'{self.make} is at position {self.pos}'

    def main():
        myCar = Car()
        print(myCar)
        myCar.move(15)
        print(myCar)

if __name__=="__main__":
    Car.main()

File: week3/lab_java/Car.java
package lab3;

import utils.In;

public class Car {
    private String make;
    private int pos;

    // Constructor
    private Car() {
        this.make = "BMW";
        this.pos = 0;
    }

    private void move(int distance) {
        this.pos += distance;
    }

    @Override
    public String toString() {
        return this.make+" is at position "+this.pos;
    }

    public static void main(String[] args) {
        Car myCar = new Car();
        System.out.printf("%s is at position: %d\n",myCar.make,myCar.pos);
        myCar.move(15);
        System.out.printf("%s is at position: %d\n",myCar.make,myCar.pos);
    }
}
